restructure pod delete (#540)
harness/developer-hub: docs/chaos-engineering/chaos-faults/kubernetes/pod/pod-delete.md

---

Pod delete is a Kubernetes pod-level chaos fault that causes specific (or random) replicas of an application resource to fail forcibly (or gracefully).

> - It simulates:
>   - Graceful delete (or rescheduling) of pods as a result of upgrades; 
>   - Forced deletion of pods as a result of eviction; and 
>   - Leader-election in complex applications.

Reviewer: ```suggestion
- It simulates:
  - Graceful delete (or rescheduling) of pods as a result of upgrades
  - Forced deletion of pods as a result of eviction
  - Leader-election in complex applications
```

Don't need semicolon or "and", per MS MOS: https://learn.microsoft.com/en-us/style-guide/scannable-content/lists#punctuation

> - In distributed systems like Kubernetes, your application replicas may not be sufficient to manage the traffic (indicated by SLIs) when certain replicas are unavailable.

Reviewer: ```suggestion
- In distributed systems, like Kubernetes, your application replicas may not be sufficient to manage the traffic (indicated by SLIs) when certain replicas are unavailable.
```

- To ensure smooth usage, is important for the applications to have a minimum number of available replicas.
- When the pressure on other replicas increases, the horizontal pod autoscaler scales based on the observed resource utilization.

![Pod Delete](./static/images/pod-delete.png)

## Use cases

> - It can be used to check the application's deployment sanity (replica availability and uninterrupted service) and recovery workflow.

Reviewer: ```suggestion
- Pod delete can be used to check the application's deployment sanity (replica availability and uninterrupted service) and recovery workflow.
```

Although the whole page is about pod delete, if someone follows an anchored link directly to this subsection, it would be helpful if the first bullet restated the chaos fault name.

- It can be used to verify:

>   - Disk (or volume) re-attachment times in stateful applications, 
>   - Application start-up times, and readiness probe configuration (health endpoints and delays)
>   - Adherence to topology constraints (node selectors, tolerations, zone distribution, affinity (or anti-affinity) policies),
>   - Proxy registration times in service-mesh environments, 
>   - Post (lifecycle) hooks and termination seconds configuration for the microservices (under active load)- i.e. graceful termination handling, 
>   - Whether resource budgeting on cluster nodes (whether request(or limit) settings are honored on available nodes for successful schedule).

Reviewer: ```suggestion
  - Disk (or volume) re-attachment times in stateful applications
  - Application start-up times, and readiness probe configuration (health endpoints and delays)
  - Adherence to topology constraints (node selectors, tolerations, zone distribution, and affinity/anti-affinity policies)
  - Proxy registration times in service-mesh environments
  - Post (lifecycle) hooks and termination seconds configuration for the microservices (under active load)- i.e. graceful termination handling
  - Whether resource budgeting on cluster nodes (whether request(or limit) settings are honored on available nodes for successful schedule)
```

Removed commas at end of each bullet, per MS MOS (link in previous comment).
I would suggest finding a way to remove the double parenthesis in the third bullet. You don't have to use my suggestion.

**Note**

> - You will need Kubernetes > 1.16 to execute this fault.

Reviewer: ```suggestion
- You need Kubernetes > 1.16 to execute this fault.
```

- Ensure that application pods are in running state before and after chaos injection.

## Fault tunables

> <h3>Optional fields</h3>

Reviewer: Are you changing this from a <details> accordion to a regular section? In that case, I suggest markdown ( `### Optional fields` ) instead of HTML.

Please add a sentence between heading levels or combine into one heading. There is only one subsection here, so I suggest:

```
## Fault tunables

You can configure the following optional fields.
```

      </tr>
      <tr>
        <td> TOTAL_CHAOS_DURATION </td>
        <td> Duration that you specify, through which chaos is injected into the target resource (in seconds).</td>

>         <td> Defaults to 15s. Overall run duration of the fault may exceed the <code>TOTAL_CHAOS_DURATION</code> by a few minutes. More information <a href = "https://developer.harness.io/docs/chaos-engineering/chaos-faults/common-tunables-for-all-faults#duration-of-the-chaos">here.</a></td>

Reviewer: ```suggestion
        <td>Defaults to 15s. Overall run duration of the fault may exceed the <code>TOTAL_CHAOS_DURATION</code> by a few minutes. For more information, go to <a href = "https://developer.harness.io/docs/chaos-engineering/chaos-faults/common-tunables-for-all-faults#duration-of-the-chaos">Duration of the chaos</a>.</td>
```

For links, please use the name of the destination page/section rather than the word "here". This lets the reader know where they are going before they click.
